accepted/global-json-updates.md

```diff
@@ -0,0 +1,374 @@
+
+```json
+{
+  "sdk": {
+    "version": "3.0.100-Pre",
+    "allowPrerelease": false,
+    "rollForward": "latestMajor"
+  }
+}
+```
+
+This will always fail. The error will be:
+
+```
+No SDK could be selected due to global.json issue: Specifying a prerelease version version with exactMatch and allowPrerelease is set to false will always fail.
+```
+
+### 7. Fail if exact match is not found
+
+If the user wants an exact match, they could use the following.
+
+```json
+{
+  "sdk": {
+    "version": "2.2.100",
+    "allowPrerelease": true,
+    "rollForward": "disable"
+  }
+}
+```
+
+The common `global.json` for this case is likely to be:
+
+```json
+{
+  "sdk": {
+    "version": "2.2.100",
+    "rollForward": "disable"
+  }
+}
+```
+
+When this fails, the error is:
+
+```
+The requested SDK version was not available, and SDK version roll-forward was disabled.
+```
+
+### 8. Highest within runtime major.minor
+
+If the user wants the highest within a runtime major/minor, they could use the following. 
+
+
+```json
+{
+  "sdk": {
+    "version": "2.2.100",
+    "allowPrerelease": true,
+    "rollForward": "latestFeature"
+  }
+}
+```
+
+### 9. Exact match or highest within runtime major.minor
+
+This proposal does not support arbitrary ranges or arbitrary upper bounds.
+
+
+### Selections made
+
+This table lists the SDKs selected in each of the above scenarios. 
+
+| Available SDKs                | #1      | #2          | #3      | #4          | #5      | #6      | #7      | #8      |
+|-------------------------------|---------|-------------|---------|-------------|---------|---------|---------|---------|
+| 2.1.700                       | fail    | 2.1.700     | 2.1.700 | fail        | fail    | fail    | fail    | fail    |
+| 2.2.100                       | 2.2.100 | 2.2.100     | 2.2.100 | 2.2.100     | 2.2.100 | fail    | 2.2.100 | 2.2.100 |
+| 2.2.103                       | 2.2.103 | 2.2.103     | 2.2.103 | 2.2.103     | 2.2.103 | fail    | fail    | 2.2.103 |
+| 2.1.700, 2.2.100, 2.2.103     | 2.2.100 | 2.2.103     | 2.2.103 | 2.2.103     | 2.2.103 | fail    | 2.2.100 | 2.2.100 |
+| 2.1.700, 2.2.103, 3.1.100-Pre | 2.2.103 | 3.1.100-Pre | 2.2.103 | 3.1.100-Pre | 2.2.103 | fail    | fail    | 2.2.100 |
+| 2.1.700, 2.2.103, 3.1.100     | 2.2.103 | 3.1.100     | 3.1.100 | 3.1.100     | 3.1.100 | fail    | fail    | 2.2.103 |
+
+## Other proposals considered
+
+Prior to arriving at this proposal, two others were considered. Neither of these align well with roll-forward on the .NET Core Runtime.
+
+* An independent versioning scheme. It's largest downside is that it would be _another_ way users would have to think about specifying versioning behavior. 
+* An attempt to parallel NuGet. Definining [NuGet version ranges](https://docs.microsoft.com/en-us/nuget/reference/package-versioning) is relatively complex, and it didn't feel like a good fit. 
```

Comment:
We may also want to consider a version blacklist.  A specific version is incompatible with a given project.  Triggers a compile exception, generate bad code ...

Reply:
@sdmaclea Is there a specific scenario, or is this for completeness/in case we screw up. 

Since we're human, this makes sense to consider for a backlog item, but I don't want to make the current round of changes any more complex unless necessary.

Reply:
This is for completeness or in case we screw up.

We don't have a way to work around specific broken SDK releases. It is especially problematic when the latest release is broken.  Presumably the preferred solution here would be to use `disable` with a specific version.